File: api/src/main/java/org/pmiops/workbench/db/dao/UserDataUseAgreementDao.java
package org.pmiops.workbench.db.dao;

import java.util.List;
import org.pmiops.workbench.db.model.DbUserDataUseAgreement;
import org.springframework.data.repository.CrudRepository;

@Deprecated() // will be replaced by UserCodeOfConductAgreementDao as part of RW-4838
public interface UserDataUseAgreementDao extends CrudRepository<DbUserDataUseAgreement, Long> {
  List<DbUserDataUseAgreement> findByUserIdOrderByCompletionTimeDesc(long userId);
}

File: api/src/main/java/org/pmiops/workbench/db/dao/UserDataUseAgreementDao.kt
package org.pmiops.workbench.db.dao

import org.pmiops.workbench.db.model.UserDataUseAgreement
import org.springframework.data.repository.CrudRepository

interface UserDataUseAgreementDao : CrudRepository<UserDataUseAgreement, Long> {
    fun findByUserIdOrderByCompletionTimeDesc(userId: Long): List<UserDataUseAgreement>
}
